components/roster/heroes/Heroes.tsx

```diff
@@ -0,0 +1,284 @@
+"use client";
+
+import { FormEvent, useReducer } from "react";
+import { Hero } from "@/lib/roster";
+import { Card, CardContent } from "@/components/ui/card";
+import { AscensionLevel } from "@/lib/characters";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuSub,
+  DropdownMenuSubContent,
+  DropdownMenuSubTrigger,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Checkbox } from "@/components/ui/checkbox";
+import { Label } from "@/components/ui/label";
+import HeroPortrait from "@/components/roster/heroes/HeroPortrait";
+import { Check, Star, Unlock, Lock } from "lucide-react";
+import Image from "next/image";
+import { rosterImages } from "@/lib/images";
+import { Button } from "@/components/ui/button";
+import { toast } from "sonner";
+import { cn } from "@/lib/utils";
+
+type SubSelectProps = {
+  hero: Hero;
+  heroProperty: "ascension" | "equipment";
+  options: AscensionLevel[] | string[];
+  selected: AscensionLevel | string | null | undefined;
+  label: string;
+  disabled: boolean;
+};
+
+type HeroProps = {
+  heroList: Hero[];
+};
+
+const isAscensionLevelValid = (level: AscensionLevel): boolean => {
+  if (!level) return false;
+  const validLevels = Object.keys(AscensionLevel).splice(7);
```

Comment:
maybe we don't `splice(7)` here to ensure that if they do add a new whale tier of ascension we can support it without a code change?

Reply:
Wait huh? This splice 7 is basically saying anything Mythic+ or over.. you think they'll change something like that?

Reply:
ah i misunderstood the order of the ascension level keys, all good